In [8]:
import pandas as pd

# Обзор и предобработка данных

In [34]:
dataset = pd.read_csv('movies.csv')

In [35]:
dataset.head(n=2)

poster  \
0  https://m.media-amazon.com/images/S/sash/4Fyxw...   
1  https://m.media-amazon.com/images/S/sash/4Fyxw...   

                      title certificate  runtime                 genre  \
0  The Shawshank Redemption          15  142 min                 Drama   
1           The Dark Knight         12A  152 min  Action, Crime, Drama   

   rating                                              about  \
0     9.3  Two imprisoned men bond over a number of years...   
1     9.0  When the menace known as the Joker wreaks havo...   

            director                                              stars  \
0     Frank Darabont  ('Tim Robbins',), ('Morgan Freeman',), ('Bob G...   
1  Christopher Nolan  ('Christian Bale',), ('Heath Ledger',), ('Aaro...   

       votes gross_earn  
0  26,26,905    $28.34M  
1  25,98,173   $534.86M

In [36]:
dataset.shape

(5000, 11)

In [37]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   poster       5000 non-null   object 
 1   title        5000 non-null   object 
 2   certificate  5000 non-null   object 
 3   runtime      5000 non-null   object 
 4   genre        4970 non-null   object 
 5   rating       5000 non-null   float64
 6   about        4986 non-null   object 
 7   director     5000 non-null   object 
 8   stars        5000 non-null   object 
 9   votes        5000 non-null   object 
 10  gross_earn   4585 non-null   object 
dtypes: float64(1), object(10)
memory usage: 429.8+ KB


In [38]:
dataset['title'].nunique(), "уникальных фильмов из", len(dataset)

(4891, 'уникальных фильмов из', 5000)

In [39]:
dataset.describe(include='all')

poster     title  \
count                                                5000      5000   
unique                                                  1      4891   
top     https://m.media-amazon.com/images/S/sash/4Fyxw...  Godzilla   
freq                                                 5000         3   
mean                                                  NaN       NaN   
std                                                   NaN       NaN   
min                                                   NaN       NaN   
25%                                                   NaN       NaN   
50%                                                   NaN       NaN   
75%                                                   NaN       NaN   
max                                                   NaN       NaN   

       certificate runtime                   genre       rating         about  \
count         5000    5000                    4970  5000.000000          4986   
unique          39     178                     346          NaN          4976   
top             15  97 min  Comedy, Drama, Romance          NaN  The story of   
freq          1843     139                     242          NaN             8   
mean           NaN     NaN                     NaN     6.692300           NaN   
std            NaN     NaN                     NaN     0.973649           NaN   
min            NaN     NaN                     NaN     1.500000           NaN   
25%            NaN     NaN                     NaN     6.100000           NaN   
50%            NaN     NaN                     NaN     6.800000           NaN   
75%            NaN     NaN                     NaN     7.400000           NaN   
max            NaN     NaN                     NaN     9.300000           NaN   

           director                                              stars  \
count          5000                                               5000   
unique         1969                                               4969   
top     Woody Allen  ('William Shatner',), ('Leonard Nimoy',), ('De...   
freq             38                                                  6   
mean            NaN                                                NaN   
std             NaN                                                NaN   
min             NaN                                                NaN   
25%             NaN                                                NaN   
50%             NaN                                                NaN   
75%             NaN                                                NaN   
max             NaN                                                NaN   

         votes gross_earn  
count     5000       4585  
unique    4918       3372  
top     35,423     $0.01M  
freq         2         27  
mean       NaN        NaN  
std        NaN        NaN  
min        NaN        NaN  
25%        NaN        NaN  
50%        NaN        NaN  
75%        NaN        NaN  
max        NaN        NaN

In [40]:
duplicates = dataset.duplicated()
print(f"Количество дубликатов: {duplicates.sum()}")


Количество дубликатов: 0


In [43]:
# Проверка пропусков
missing_data = dataset.isnull().sum()
print(f"Пропущенные значения:\n{missing_data}")

Пропущенные значения:
poster           0
title            0
certificate      0
runtime          0
genre           30
rating           0
about           14
director         0
stars            0
votes            0
gross_earn     415
dtype: int64


# Вывод
1. Размеры датасета:
    - Датасет содержит 5000 записей и 11 признаков;
    - Всего в датасете представлено 5000 записей, из которых 4891 уникальных;
2. Типы данных:
    - Основные данные представлены в виде строковых (object) и числовых (float64) типов.
    - Некоторые столбцы требуют приведения типов:
        - runtime (время) — преобразовать в целочисленный формат;
        - votes (голоса) — преобразовать из строки с запятыми в числовой формат;
        - gross_earn (доход) — содержит символы $ и запятые, необходимо привести к числовому формату.
3. Дубликаты:
    - Полных дубликатов нет;
    - Выявлено 109 дубликатов, которые составляют 2,18% от общего числа записей.
4. Пропущенные значения:
    - Пропущенные значения в столбцах:
        - genre (жанр) - 30 пропущенных значений (0,6%);
        - about - 14 пропущенных значений (0,28%);
        - gross_earn - 415 пропущенных значений (8,3%)
    - Пропуски в `genre` и `about` можно заполнить строковым значением 'Unknown' или удалить, если их доля мала.
    - Для gross_earn необходимо заполнить или удалить строки.
5. Возможные проблемы при анализе:
    - Неправильные типы данных (например, votes и gross_earn) могут привести к ошибкам в расчетах.
    - Пропущенные значения в gross_earn могут исказить результаты анализа доходов фильмов.
    - Некоторые строковые признаки, такие как genre, требуют дополнительной обработки для анализа (например, разбиение на подкатегории).
    - Малая доля пропусков в about и genre вряд ли критична, но требует внимания.